conans/client/build/compiler_id.py

```diff
@@ -0,0 +1,211 @@
+# -*- coding: utf-8 -*-
+
+import os
+import tempfile
+from six import StringIO
+
+from conans.client.runner import ConanRunner
+
+
+GCC = "gcc"
+LLVM_GCC = "llvm-gcc"  # GCC frontend with LLVM backend
+CLANG = "clang"
+APPLE_CLANG = "apple-clang"
+SUNCC = "suncc"
+MSVC = "Visual Studio"
+INTEL = "intel"
+QCC = "qcc"
+
+
+class CompilerId:
```

Comment:
This class can benefit from an inheritance from the ``Version`` class and avoid duplicating the comparison logic

Reply:
/cc @jgsogo 
inheritance from the `Version` class appears to be problematic, I can't get it working:
```
    UNKNOWN_COMPILER = CompilerId(None, None, None, None)
TypeError: __new__() takes 2 positional arguments but 5 were given
```
I'll use composition instead

Reply:
Yes, that `Version` class inherits from `str` (and inmutable object) and then it requires the `__new__` meta-constructor. IMO it is not the best implementation and we should refactor `Version` and these new `CompilerId` class, but this work can be done in another PR after implementing this feature. It should be quite straightforward.

BTW, @SSE4 , you can use `total_ordering` from `functools` to implement all those `eq`, `lt`, `gt`,... to compare objects.

Reply:
right now I don't think I need ordering comparison, only equality is required

Reply:
used Version to avoid duplication (without inheritance)

Reply:
For what I see, this class should extend a namedtuple to avoid duplicated code.